src/utils/time.ts

```diff
@@ -59,6 +77,16 @@ export const wait = async (timeMilli: number): Promise<void> => {
   });
 };
 
+/**
+ * This function returns a promise that resolves after the specified time interval and throws
+ * if the promise passed as an argument does not resolve before the specified time interval.
+ *
+ * @throws An error if the promise passed as an argument does not resolve before the specified time interval.
+ *
+ * @param promise - The promise to be resolved.
+ * @param timeoutMs - The time interval in milliseconds.
```

Comment:
Not sure the @throws is correctly placed relative to other TSDoc tags, read the notion.
```suggestion
 * @param timeoutMs - The time interval in milliseconds.
 
```